### ISSUE TITLE
Support for SASL PLAIN authentication for Collector Kafka Storage Backend

### ISSUE THREAD

**Issue by somnathnitb** [jaegertracing/jaeger#1845]
Kafka supports different types of authentication mechanisms. As of today collector for Kafka storage backend only supports SASL GSSPI (Kerberos) and certificate-based authentication (mutual TLS). However, not all Kafka deployments support this authentication mechanism. 
In our case, we have a Kafka cluster deployed with TLS transport for communication however our authentication is set up to use SASL PLAIN https://docs.confluent.io/current/kafka/authentication_sasl/authentication_sasl_plain.html#kafka-sasl-auth-plain. Can we add support for SASL PLAIN authentication for the collector with Kafka as storage backend?

**Comment by jpkrohling** [comment]
Yes, if it's only adding a new auth mechanism to an existing storage backend, it will certainly be accepted! Check [CONTRIBUTING.md](https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING.md) for more info on what's expected from your PR.

**Comment by somnathnitb** [comment]
@jpkrohling how can I get contributor access here so that I can push my branch and raise PR?

**Comment by yurishkuro** [comment]
Please fork this repo, create a branch in the fork, and then you'll be able to create a pull request from that branch. See https://help.github.com/en/articles/about-pull-requests
